### PR TITLE
Add tooltips to `Hide comment type` settings where necessary

### PR DESCRIPTION
**Pull request by delvh**
Previously, this setting was pretty confusing for users, especially the difference between "reference" and "issue reference".
Related: #21321.

### CODE REVIEW

**Comment by delvh** [comment]
Review hint: Hide whitespace changes.

**Comment by lunny** [comment]
image
The position of the tooltip is wired.

**Comment by delvh** [comment]
I didn't change the positioning of the tooltip.
As far as I know, tippy automatically detects whether to position on the top or the bottom depending on where you hover over the box.
Meaning when you enter the box from the bottom and then hover there, it should be positioned on the bottom.
Or it positions it on the top otherwise.

**Comment by silverwind** [comment]
Tippy may sometimes try to be smart about positioning, but you can add `data-tooltip-placement="top"` to force it on top.

**Comment by delvh** [comment]
ping for maintainers.
This PR is up to date, and would be nice to have as it helps against the current confusion.

**Comment by lunny** [comment]
I couldn't know why we need a tooltip here. Maybe just some text on top of all checkboxes is good enough?

image

Like this
image

**Comment by delvh** [comment]
> Related: https://github.com/go-gitea/gitea/issues/21321.

That's exactly my point why I opened this issue.
There's simply no guideline yet on when to use what.
I agree, for the topmost tooltip, a static help text is probably the best.
But not for the ones below.

**Comment by delvh** [comment]
@lunny applied.

File: templates/user/settings/appearance.tmpl [hunk 1]
```diff
@@ -70,101 +70,106 @@
...
-					<div class="ui checkbox">
-						<input name="reference" type="checkbox" {{if(call .IsCommentTypeGroupChecked "reference")}}checked{{end}}>
-						<label>{{.locale.Tr "settings.comment_type_group_reference"}}</label>
+		<div class="ui header">
+			<h4 class="ui top attached header">
+				{{.locale.Tr "settings.hidden_comment_types"}}
+			</h4>
+			<div class="ui attached segment">
+				<p class="help">
+					{{.locale.Tr "settings.hidden_comment_types_tooltip"}}
```
Comment (wxiaoguang) [inline templates/user/settings/appearance.tmpl]: It's not a tooltip now, so the locale key could have a new name?

File: templates/user/settings/appearance.tmpl [hunk 2]
```diff
@@ -70,101 +70,106 @@
...
-			{{.locale.Tr "settings.hidden_comment_types"}}
-		</h4>
-		<div class="ui attached segment">
-			<form class="ui form" action="{{.Link}}/hidden_comments" method="post">
-				{{.CsrfTokenHtml}}
-				<div class="inline field">
-					<div class="ui checkbox">
-						<input name="reference" type="checkbox" {{if(call .IsCommentTypeGroupChecked "reference")}}checked{{end}}>
-						<label>{{.locale.Tr "settings.comment_type_group_reference"}}</label>
+		<div class="ui header">
```
Comment (wxiaoguang) [inline templates/user/settings/appearance.tmpl]: Does this new `<div class="ui header">` have a special purpose?

There is already `<h4 class="ui top attached header">` below.

And I do not think put `<div class="ui attached segment">` inside a `header` is correct layout.

ps: Fomantic UI has its best practices, I guess many UI problems are caused by misusage on Gitea side, Fomantic UI is innocent in many cases 🤣

**Review by lunny — APPROVED** [review]
Have a test with no problem from UI.

**Comment by delvh** [comment]
@wxiaoguang @lunny all done.

**Review by HesterG — APPROVED** [review]

**Pull request merged**